### DISCUSSION TITLE
macOS autostart (start on login)

### DISCUSSION THREAD

**Discussion by mrienstra** [ActivityWatch/activitywatch#950] [Ideas]
According to https://docs.activitywatch.net/en/latest/getting-started.html#autostart:
> Autostart is set up automatically by the Windows installer and for Arch Linux by the AUR package (if your desktop environment supports [XDG Autostart](https://wiki.archlinux.org/index.php/XDG_Autostart)).

Would be kind of handy to add this for macOS as well, though of course it's a fairly low-priority, as it's not hard to do manually.

This project may be able to use the same strategy that https://github.com/leits/MeetingBar did:

- https://github.com/leits/MeetingBar/issues/494#issuecomment-1463946780
- https://github.com/leits/MeetingBar/commit/76b3c267af41dc72ffd5b713ae70955322a5607d
- https://github.com/sindresorhus/LaunchAtLogin

**Comment by ErikBjare** [comment]
I'd be happy to accept PRs for this!

Although not sure how to do it, given that those methods assume a Swift project.

**Reply by mrienstra** [reply]
Prior to that, this is the approach they were using:
https://jogendra.dev/implementing-launch-at-login-feature-in-macos-apps

... but that might be obsolete, and is also using Swift!

Well, maybe someone else with the proper subject matter expertise will find this and weigh in on the right approach to take.
